tests/test_net_connect.js

```diff
@@ -88,8 +86,33 @@ describe('`enableNetConnect()`', () => {
 
     await assertRejects(
       got('https://example.test/'),
-      Error,
-      'Nock: Disallowed net connect for "example.test:80/"'
+      /Nock: Disallowed net connect for "example.test:443\/"/
+    )
+  })
+
+  it('enables real HTTP request only for specified domain, via function', async () => {
+    const onResponse = sinon.spy()
+    const server = http.createServer((request, response) => {
+      onResponse()
+      response.writeHead(200)
+      response.end()
+    })
+    await new Promise(resolve => server.listen(resolve))
+
+    nock.enableNetConnect(host => host.includes('ocalhos'))
+
+    await got(`http://localhost:${server.address().port}/`)
+    expect(onResponse).to.have.been.calledOnce()
+
+    server.close()
+  })
+
+  it('disallows request for other domains, via function', async () => {
+    nock.enableNetConnect(host => host.includes('ocalhos'))
+
+    await assertRejects(
+      got('https://example.test/'),
+      /Nock: Disallowed net connect for "example.test:443\/"/
```

Comment:
One other comment: to be really thorough, you could use a sinon stub for your matcher and assert that it’s being called with the exact right argument.